src/devices/Ssd1351/Ssd1351Commands.cs

```diff
@@ -0,0 +1,541 @@
+        /// RAM. If vertical address increment mode is enabled by command A0h, after finishing 
+        /// read/write one row data, it is incremented automatically to the next row address.
+        /// Whenever the row address pointer finishes accessing the end row address, it is 
+        /// reset back to start row address.
+        /// </summary>
+        /// <param name="startRow">Row start address with a range of 0-127.</param>
+        /// <param name="endRow">Row end address with a range of 0-127.</param>
+        public void SetRowAddress(byte startRow = 0x00, byte endRow = 0x7F)
+        {
+            if (startRow > 0x7F)
+            {
+                throw new ArgumentException("The row start address is invalid.", nameof(startRow));
+            }
+
+            if (endRow > 0x7F)
+            {
+                throw new ArgumentException("The row end address is invalid.", nameof(endRow));
+            }
+
+            SendCommand(Ssd1351Command.SetRow, startRow, endRow);
+        }
+
+        /// <summary>
+        /// This command sets the divide ratio to generate DCLK (Display Clock) from CLK and
+        /// programs the oscillator frequency Fosc that is the source of CLK if CLS pin is pulled high.
+        /// </summary>
+        /// <param name="displayClockDivideRatio">Display clock divide ratio with a range of 0-15.</param>
+        /// <param name="oscillatorFrequency">Oscillator frequency with a range of 0-15.</param>
+        public void SetDisplayClockDivideRatioOscillatorFrequency(byte displayClockDivideRatio = 0x01, byte oscillatorFrequency = 0x0D)
+        {
+            if (displayClockDivideRatio > 0x0F)
+            {
+                throw new ArgumentException("The display clock divide ratio is invalid.", nameof(displayClockDivideRatio));
+            }
+
+            if (oscillatorFrequency > 0x0F)
+            {
+                throw new ArgumentException("The oscillator frequency is invalid.", nameof(oscillatorFrequency));
+            }
+
+            SendCommand(Ssd1351Command.SetClockDiv, (byte)((oscillatorFrequency << 4) | displayClockDivideRatio));
+        }
+
+        /// <summary>
+        /// This command is used to set Contrast Setting of the display.
+        /// The chip has 256 contrast steps from 00h to FFh.  The segment
+        /// output current ISEG increases linearly with the contrast step,
+        /// which results in brighter display.
+        /// </summary>
+        /// <param name="colorAContrast">Contrast level for color A.</param>
+        /// <param name="colorBContrast">Contrast level for color B.</param>
+        /// <param name="colorCContrast">Contrast level for color C.</param>
+        public void SetContrastABC(byte colorAContrast = 0x86, byte colorBContrast = 0x51, byte colorCContrast = 0x86)
+        {
+            SendCommand(Ssd1351Command.SetContrastABC, colorAContrast, colorBContrast, colorCContrast);
```

Comment:
Shouldn't we be doing parameter checks here?

Reply:
The contrast values are 0->255 and are the range of the byte parameters.